pages/treasury/create-proposal.tsx

```diff
@@ -176,21 +233,39 @@ const CreateProposal = () => {
             onChange={(e) => {
               setFormTitle(e.target.value);
             }}
-          />
-          <TextArea
-            name="description"
-            value={formDescription}
-            onChange={(e) => {
-              setFormDescription(e.target.value);
-            }}
             css={{
-              mt: "$5",
-              height: 360,
-              // "boxShadow:active": "$colors$primary8 inset 0px 0px 0px 1px",
+              marginBottom: "$5",
             }}
-            placeholder="Describe your proposal (markdown)"
-            size="3"
           />
+
+          <Tabs>
+            <TabList>
+              <StyledTab>Write</StyledTab>
+              <StyledTab>Preview</StyledTab>
+            </TabList>
+            <TabPanels>
+              <TabPanel>
+                <TextArea
+                  name="description"
+                  value={formDescription}
+                  onChange={(e) => {
+                    setFormDescription(e.target.value);
+                  }}
+                  css={{
+                    height: 360,
+                    borderTopLeftRadius: "0",
+                  }}
+                  placeholder="Describe your proposal (markdown)"
+                  size="3"
+                />
+              </TabPanel>
+              <TabPanel style={{ paddingBottom: "4px" }}>
+                <MarkdownPreview remarkPlugins={[remarkGfm]}>
+                  {formDescription}
```

Comment:
We should probably use `description` here instead, which is what we actually set for the on-chain proposal created. It merges the title with the content to get a single markdown and is what we display on the proposal details page as well: https://github.com/livepeer/explorer/blob/48f49db5ac464cd05891dd4268c32a2b5ddadece/pages/treasury/create-proposal.tsx#L82

That should avoid having people commit the mistake of repeating the markdown `#` title inside the description of the proposal, which then gets displayed duplicated like the Fanteasy recent proposal.

(in general though I don't really love this UI to set the title separately from the "content" and then merge them into the same thing, but couldn't think of a way to enforce we always have a title here)

Reply:
Hi @victorges,

Thank you for your review. I decided not to use the `description` variable because it returns `null` if either the title or description is not set. This seems to be by design so that the pipeline only goes through when both fields are filled in.

Here’s the relevant code:

https://github.com/livepeer/explorer/blob/b37f5077b1d6511832c690e3e5784ef1fbf03432/pages/treasury/create-proposal.tsx#L136

I have now added the title to the preview:

https://github.com/livepeer/explorer/pull/267/commits/d241342682d4b72463eff331bcd03e74c51b07ff

Would you prefer that I rewrite the codebase to use the description directly instead?

Reply:
Got it! Thanks for explaining it here. No need to rewrite, LGTM!